pkg/k8s/types.go

```diff
@@ -111,8 +111,4 @@ type Controller interface {
 	// UpdateStatus updates the status subresource for the given resource and GroupVersionKind
 	// The object within the 'interface{}' must be a pointer to the underlying resource
 	UpdateStatus(interface{}) (metav1.Object, error)
-
-	// K8sServiceToMeshServices translates a k8s service with one or more ports to one or more
-	// MeshService objects per port.
-	K8sServiceToMeshServices(corev1.Service) []service.MeshService
```

Comment:
I'd have preferred to keep this here since this is very much related to the k8s.Controller implementation. What is the reason to remove this?

Reply:
I tried to explain this in the commit message, but this doesn't need to be part of the interface assuming it only depends on other methods in the interface. That gives me the impression this method would be exactly the same for any type implementing the interface. If that's true, then making it a function instead which accepts the interface lets us reuse the functionality between all types implementing the interface.

Reply:
I applaud @nojnhuh in taking the initiative to simplify the interface.  Agree with Jon - there is nothing special about this transformer, not sure it belongs to this interface given that we already have `GetEndpoints` and `ListServices`